databrowser/src/pages/datasets/overview/OverviewListPage.vue

```diff
@@ -366,12 +383,50 @@ const hideFilters = () => {
   isFiltersModalVisible.value = false;
 };
 
+const filterSelectedForComponent = ref<{ key: string; value: string }[]>([]);
+
+const initializeFiltersAndSearch = () => {
+  const {rawfilter, searchQuery} = getStartedQuery();
+
+  if (!searchQuery && !rawfilter) {
+    return;
+  }
+
+  if (searchQuery && searchQuery !== '') {
+    filters.value.searchVal = (searchQuery as string) || '';
```

Comment:
`searchQuery` is already known to be a string, no need to cast it to `string` again.

In addition, here I'm not 100% sure what you try to express with `... || ''`, `searchQuery` is typed as `string`, so the `||` will never be evaluated.

If `searchQuery` can be `undefined`, `null` or something else, then the types are not correct and you need to fix them.